### PR TITLE
Modify envelopes and test

### PR DESCRIPTION
**Pull request by JanosJiri**
Small modifications.

- removing redundant loop for `sin` and `sin2` envelopes
- modifying `envelopes_analysis` so that it takes the functions from `promdens.py` (we would see problems in envelopes visually with that)
- changing threshold for tests of pda and pdaw

### CODE REVIEW

File: src/promdens/promdens.py [hunk 1]
```diff
@@ -111,24 +111,23 @@ def __post_init__(self):
 
         if self.envelope_type == 'gauss':
             self.equation = "exp(-2*ln(2)*(t-t0)^2/fwhm^2)*cos((omega+chirp*t)*t)"
-            extent = 2.4 * self.fwhm
+            extent = 2.4*self.fwhm
```
Comment (danielhollas) [inline src/promdens/promdens.py:114]: Feel free to rename this to temporal_extent. (Although I guess this is half of that,  not sure how to express that )

File: src/promdens/promdens.py [hunk 2]
```diff
@@ -146,32 +145,21 @@ def calc_field_envelope(self, t: np.ndarray) -> np.ndarray:
...
-
             return 1/np.cosh(2*np.log(1 + np.sqrt(2))*(t - self.t0)/self.fwhm)
-
         elif self.envelope_type == 'sin':
-
             field = np.zeros(shape=np.shape(t))
-            for k in range(np.shape(t)[0]):
-                if t[k] >= self.tmin and t[k] <= self.tmax:
-                    field[k] = np.sin(np.pi/2*(t[k] - self.t0 + self.fwhm)/self.fwhm)
+            trange = np.logical_and(t >= self.tmin,t <= self.tmax)
```
Comment (danielhollas) [inline src/promdens/promdens.py:155]: Wow, nice, fancy :-) Assuming you checked visually that this works.

File: supplementary/envelopes_analysis/field_envelopes.py [hunk 3]
```diff
@@ -1,163 +1,18 @@
...
-        return W
-
+from promdens.promdens import LaserPulse, InitialConditions, ENVELOPE_TYPES
 
 # input data
 fwhm = 10  # in fs
 omega = 1  # in a.u.
 lchirp = 0  # in a.u.
 t0 = 0  # in fs
-envelope_types = ['sin', 'sin2', 'gauss', 'sech', 'lorentz']
```
Comment (danielhollas) [inline supplementary/envelopes_analysis/field_envelopes.py:160]: Let's leave this list here, it's much easier to understand that the sorting below.

**Review by danielhollas — APPROVED** [review]
Cool! Just some minor suggestions.

File: src/promdens/promdens.py (hunk 1 above)
Comment (JanosJiri) [reply src/promdens/promdens.py:114]: I forgot to do that, thanks. I will change it.

**Pull request merged**